### PR TITLE
NXP-23706 testing H2 only

### PR DESCRIPTION
**Pull request by andreinechaev**

### CODE REVIEW

File: nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryFulltextQuery.java [hunk 1]
```diff
@@ -527,8 +527,13 @@ public void testFulltextPrefix() throws Exception {
         }
     }
 
+    /**
+     * We want fulltext queries work only for H2, assuming that it's a development environment
+     * and we still want to perform these queries. For instance for our REST calls */
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryFulltextQuery.java]: Format

File: nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryFulltextQuery.java [hunk 2]
```diff
@@ -527,8 +527,13 @@ public void testFulltextPrefix() throws Exception {
...
     @Test
-    public void testWildCardsInFulltextSearch() throws Exception {
+    public void testWildcardsInFulltextSearchForH2() throws Exception {
+        if (!coreFeature.getStorageConfiguration().isVCSH2()) { return; }
```
Comment (kevinleturc) [inline nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryFulltextQuery.java]: Please use assume instead, you have example in this test class such as:
```
assumeTrue("DBS cannot remove spurious characters in fulltext search", !isDBS());
```

**Comment by efge** [comment]
Given what I explained to you, the fact that Elasticsearch should be used instead of pushing features to database-level fulltext searches (which probably can't work anyway on many databases), I'm 👎 on this whole ticket unless I can see a good reason.

**Comment by efge** [comment]
See comments in https://jira.nuxeo.com/browse/NXP-23706

**Pull request closed without merging**